azkaban-common/src/main/java/azkaban/project/FlowLoader.java

```diff
@@ -0,0 +1,139 @@
+/*
+* Copyright 2017 LinkedIn Corp.
+*
+* Licensed under the Apache License, Version 2.0 (the “License”); you may not
+* use this file except in compliance with the License. You may obtain a copy of
+* the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an “AS IS” BASIS, WITHOUT
+* WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+* License for the specific language governing permissions and limitations under
+* the License.
+*/
+
+package azkaban.project;
+
+import azkaban.flow.CommonJobProperties;
+import azkaban.flow.Flow;
+import azkaban.project.validator.ValidationReport;
+import azkaban.utils.Props;
+import java.io.File;
+import java.io.FileFilter;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Loads flows from project directory.
+ */
+public abstract class FlowLoader {
+
+  protected Map<String, Flow> flowMap = new HashMap<>();
+  protected Set<String> errors = new HashSet<>();
+
+  /**
+   * Returns errors caught when loading flows.
+   *
+   * @return Set of error strings.
+   */
+  public Set<String> getErrors() {
+    return this.errors;
+  }
+
+  /**
+   * Loads the project and returns the validation report.
+   *
+   * @param project the project
+   * @param projectDir the project dir
+   * @return the validation report
+   */
+  public ValidationReport loadProject(final Project project, final File projectDir) {
+    loadProjectFlow(project, projectDir);
+    checkJobProperties(project);
+    final ValidationReport report = new ValidationReport();
+    report.addErrorMsgs(this.errors);
+    return report;
+  }
+
+  /**
+   * Adds email properties to a flow.
+   *
+   * @param flow the flow
+   * @param prop the prop
+   */
+  public void addEmailPropsToFlow(final Flow flow, final Props prop) {
```

Comment:
This part is the same. Just moved the original code to a new method.

Reply:
Thanks for the comment.